boa/src/syntax/parser/mod.rs

```diff
@@ -83,27 +118,133 @@ impl Parser {
         }
     }
 
-    fn parse_function_parameters(&mut self) -> Result<Vec<Expr>, ParseError> {
+    /// consume the next token and increment position
+    fn get_next_token(&mut self) -> Result<Token, ParseError> {
+        self.pos += 1;
+        self.get_token(self.pos)
+    }
```

Comment:
Found the parsing bug. The problem is that we are getting the token after incrementing the pos variable. So we are always offset by one in the token stream. Now it parses the `console.log("test")` correctly.
```suggestion
    fn get_next_token(&mut self) -> Result<Token, ParseError> {
        let token = self.get_token(self.pos);
        self.pos += 1;
        token
    }
```

Reply:
haha wow was that it?

Reply:
I found a bug in the interpreter:
The code `console.log("hello");` should run. But the interpreter panics.
```text
thread 'main' panicked at 'not yet implemented', boa/src/exec/mod.rs:515:18
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```
Even though the AST is correct:
```
[boa/src/lib.rs:58] &expr = StatementList(
    [
        Call(
            GetConstField(
                Local(
                    "console",
                ),
                "log",
            ),
            [
                Const(
                    String(
                        "hello",
                    ),
                ),
            ],
        ),
    ],
)
```

Reply:
> haha wow was that it?

That was what caused the code `console.log("test")` to not parse.